pkg/nfd-worker/nfd-worker-internal_test.go

```diff
@@ -374,9 +372,6 @@ func TestCreateFeatureLabels(t *testing.T) {
 			})
 		})
 		Convey("When fake feature source is configured with a whitelist that doesn't match", func() {
-			fakeFeatureSource := source.FeatureSource(new(fake.Source))
```

Comment:
so when you removed the `recover()` from `getFeatureLabels()` you hit a panic as the fake Source did not have its config initialized right ?

Reply:
Yup, exactly